app.py

```diff
@@ -0,0 +1,362 @@
+    user_id = db.Column(db.Integer, db.ForeignKey("user.id"))
+    user = db.relationship("User")
+
+
+class Notes(db.Model):
+    __tablename__ = "notes"
+    id = db.Column(db.Integer, primary_key=True)
+    description = db.Column("description", db.String, nullable=False)
+    text = db.Column("text", db.String(500), nullable=False)
+    photo = db.Column(db.String(20), nullable=False)
+    category = db.Column("category", db.String(
+        20), nullable=False, default="Other")
+    date = db.Column(db.DateTime, nullable=True, default=datetime.now())
+    user_id = db.Column(db.Integer, db.ForeignKey("user.id"))
+    user = db.relationship("User")
+
+
+# ************************************* USER REGISTRATION / LOGIN ********************
+
+
+@app.route("/register", methods=['GET', 'POST'])
+def register():
+    db.create_all()
+    if current_user.is_authenticated:
+        return redirect(url_for('index'))
+    form = forms.RegistrationForm()
+    if request.method == 'POST' and form.validate_on_submit():
+        crypted_password = bcrypt.generate_password_hash(
+            form.password.data).decode('utf-8')
+        user = User(user=form.user.data, email=form.email.data,
+                    password=crypted_password)
+        try:
+            db.session.add(user)
+            db.session.commit()
+        except IntegrityError:
+            db.session.rollback()
+            flash('Email already registered. Please choose another.', 'danger')
+            return redirect(url_for('register'))
+        flash('Success! Now you can login', 'success')
+        return redirect(url_for('index'))
+    return render_template('register.html', title='Register', form=form)
+
+
+@app.route("/login", methods=['GET', 'POST'])
+def login():
+    db.create_all()
+    if current_user.is_authenticated:
+        return redirect(url_for('index'))
+    form = forms.LoginForm()
+    if form.validate_on_submit():
+        user = User.query.filter_by(
+            email=form.email.data).first()
+        if user and bcrypt.check_password_hash(user.password, form.password.data):
+            login_user(user, remember=form.remember_me.data)
+            next_page = request.args.get('next')
+            return redirect(next_page) if next_page else redirect(url_for('index'))
+        else:
+            flash('Something wrong. Please check email and password!', 'danger')
+    return render_template('login.html', title='login', form=form)
+
+
+@app.route("/logout")
+def logout():
+    logout_user()
+    return redirect(url_for('index'))
+
+
+@login_manager.user_loader
+def load_user(user_id):
+    db.create_all()
+    return User.query.get(int(user_id))
+
+
+@app.route("/")
+def index():
+    if current_user.is_authenticated:
+        categories = Category.query.filter_by(user_id=current_user.id).all()
+        return render_template("index.html", categories=categories, user=current_user.user)
+    else:
+        return render_template("index.html")
```

Comment:
Nereikia rašyti else, jeigu naudojame return. Vis tiek iki to kodo "nenueisime".

```suggestion
    if current_user.is_authenticated:
        categories = Category.query.filter_by(user_id=current_user.id).all()
        return render_template("index.html", categories=categories, user=current_user.user)
    return render_template("index.html")
```